govcd/edgegateway.go

```diff
@@ -1199,7 +1278,40 @@ func (egw *EdgeGateway) GetVnicIndexByNetworkNameAndType(networkName, networkTyp
 	if err != nil {
 		return nil, fmt.Errorf("cannot retrieve vNic configuration: %s", err)
 	}
-	return GetVnicIndexByNetworkNameAndType(networkName, networkType, vnics)
+	return getVnicIndexByNetworkNameAndType(networkName, networkType, vnics)
+}
+
+// GetAnyVnicIndexByNetworkName returns *int of vNic index and network type by network name
+// networkName cannot be empty
+// networkType will be one of: 'internal', 'uplink', 'trunk', 'subinterface'
+//
+// Warning: this function assumes that there are no duplicate network names attached. If it is so
```

Comment:
What we can do about it?

Reply:
In fact I could not make this happen during my tests, but still left this note as this is how the search algorithm behaves in the case it had it. The unit test `Test_getVnicIndexFromNetworkNameType` also has a fake XML to test it does so.